### PR TITLE
Bringing all the changes done in the summer sprint. [without transforms]

### PR DESCRIPTION
**Pull request by arunk-s**
Notable changes include:
- `http_request_translator` usable as a python package and as a command line tool.
-  Inclusion of tests ( current coverage 67%) and docs
- Covering of many edge cases
- Much nicer code structure (less redundancy)
- Outputting the request as original as it can be (No header order reshuffling, removal of spaces etc.)
- All the scripts uses `libcurl` in some sense to keep the request unmodified. 
- Nicer interactive session

### CODE REVIEW

File: http_request_translator/plugin_manager.py [hunk 1]
```diff
@@ -1,20 +1,51 @@
...
+    elif script_name == 'php':
+        return PHPScript
+    elif script_name == 'python':
+        return PythonScript
+    elif script_name == 'ruby':
+        return RubyScript
+    raise ValueError("The '%s' language is not supported yet." % script_name)
+
+
+def plugin_manager(script_list, parsed_tuple, search_string=None):
```
Comment (DePierre) [inline http_request_translator/plugin_manager.py]: Is it used anywhere? Doesn't look like so better remove it maybe?

**Comment by DePierre** [comment]
It looks like http_translator_request/default.py isn't used anywhere. Could you remove it?

File: http_request_translator/plugin_manager.py (hunk 1 above)
Comment (arunk-s) [inline http_request_translator/plugin_manager.py]: Yeah. Will be removed in next commit.

**Comment by arunk-s** [comment]
Yeah. I'll remove it in the next commit.

**Comment by arunk-s** [comment]
@DePierre , Removed `default.py` and necessary code.
